cmd/commands/runtime.go

```diff
@@ -440,49 +451,63 @@ func ensureIngressClass(ctx context.Context, opts *RuntimeInstallOptions) error
 	var isValidClass bool
+
 	for _, ic := range ingressClassList.Items {
-		if ic.ObjectMeta.Labels["app.kubernetes.io/name"] == "ingress-nginx" {
-			ingressClassNames = append(ingressClassNames, ic.Name)
-			ingressClassNameToController[ic.Name] = fmt.Sprintf("%s-controller", getControllerName(ic.Spec.Controller))
-			if opts.IngressClass == ic.Name {
-				isValidClass = true
+		for _, controller := range supportedControllers {
+			if ic.Spec.Controller == string(controller) {
+				ingressClassNames = append(ingressClassNames, ic.Name)
+				ingressClassNameToController[ic.Name] = ingressController{
+					Name: getIngressControllerName(controller, ic.Name),
+					Type: controller,
+				}
+
+				if opts.IngressClass == ic.Name { //if ingress class provided via flag
+					isValidClass = true
+				}
+				break
 			}
 		}
 	}
 
-	if opts.IngressClass != "" { //if user provided ingress class by flag
-		if isValidClass {
-			opts.IngressController = ingressClassNameToController[opts.IngressClass]
-			return nil
+	if opts.IngressClass != "" { //if ingress class provided via flag
+		if !isValidClass {
+			return fmt.Errorf("ingress class '%s' is not supported", opts.IngressClass)
 		}
-		return fmt.Errorf("ingress class '%s' is not supported. only the ingress class of type nginx is supported.", opts.IngressClass)
-	}
-
-	if len(ingressClassNames) == 0 {
-		return fmt.Errorf("no ingress classes of type nginx were found. please install a nginx ingress controller on your cluster before installing a runtime.")
-	}
-
-	if len(ingressClassNames) == 1 {
+	} else if len(ingressClassNames) == 0 {
+		return fmt.Errorf("no ingress classes of the supported types were found")
+	} else if len(ingressClassNames) == 1 {
 		log.G(ctx).Info("Using ingress class: ", ingressClassNames[0])
 		opts.IngressClass = ingressClassNames[0]
-		opts.IngressController = ingressClassNameToController[opts.IngressClass]
-		return nil
+	} else if len(ingressClassNames) > 1 {
+		if !store.Get().Silent {
+			err = getIngressClassFromUserSelect(ctx, ingressClassNames, &opts.IngressClass)
+			if err != nil {
+				return err
+			}
+		} else {
+			return fmt.Errorf("there are multiple ingress controllers on your cluster, please add the --ingress-class flag and define its value")
+		}
 	}
 
-	if !store.Get().Silent {
-		err = getIngressClassFromUserSelect(ctx, ingressClassNames, &opts.IngressClass)
-		if err != nil {
-			return err
-		}
+	opts.IngressController = ingressClassNameToController[opts.IngressClass].Name
+	opts.IngressControllerType = ingressClassNameToController[opts.IngressClass].Type
 
-		opts.IngressController = ingressClassNameToController[opts.IngressClass]
-		return nil
-	}
+	return nil
+}
 
-	return fmt.Errorf("please add the --ingress-class flag and define its value")
+func getIngressControllerName(controllerType ingressControllerType, className string) string {
+	switch controllerType {
+	case NginxCommunity:
+		return fmt.Sprintf("%s-controller", string(NginxCommunity))
```

Comment:
so isn't that just `k8s.io/ingress-nginx-controller`?

Reply:
yes, `ingress-nginx-controller` you think it better to just write it as it is?

Reply:
i think so. why have a type which is basically a string, and then cast it to string, and into a format-string?